manim/mobject/three_dimensions.py

```diff
@@ -189,3 +202,306 @@ def generate_points(self):
         Cube.generate_points(self)
         for dim, value in enumerate(self.dimensions):
             self.rescale_to_fit(value, dim, stretch=True)
+
+
+class Cone(ParametricSurface):
+    def __init__(
+        self,
+        base_radius=1,
+        height=1,
+        direction=Z_AXIS,
+        show_base=False,
+        resolution=24,
+        # v will stand for phi
+        v_min=0,
+        v_max=TAU,
+        # u will stand for r
+        u_min=0,
+        # u_max is calculated as a property
+        checkerboard_colors=False,
+        **kwargs
+    ):
+        self.base_radius = base_radius
+        self.height = height
+        self.show_base = show_base
+        self.direction = direction
+        ParametricSurface.__init__(
+            self,
+            self.func,
+            v_min=v_min,
+            v_max=v_max,
+            u_min=u_min,
+            u_max=np.sqrt(base_radius ** 2 + height ** 2),
+            checkerboard_colors=checkerboard_colors,
+            **kwargs
+        )
+        # used for rotations
+        self._current_theta = 0
+        self._current_phi = 0
+
+        if self.show_base:
+            self.base_circle = Dot(
+                point=self.height * IN,
+                radius=self.base_radius,
+                color=self.fill_color,
+                fill_opacity=self.fill_opacity,
+            )
+            self.add(self.base_circle)
+
+        self._rotate_to_direction()
+
+    """
+	A Cone can be defined using 2 parameters: its height, and its base radius.
+	The polar angle, theta, can be calculated using arctan(base_radius / height)
+	The spherical radius, r, can be calculated using the pythagorean theorem.
+	.. code-block::
+				   |\\
+				   |_\\ <-- theta
+		height --> |  \\
+				   |   \\ <-- r
+				   |    \\
+				   |     \\
+				   --------
+				   base_radius
+	"""
```

Comment:
Wonderful documentation/diagram, but misplaced... put in docstring/documentation.
```suggestion
```